### PR TITLE
Create MessagingProvider poller method with support for filter properties

### PR DESCRIPTION
**Pull request by jose-correia**
This PR adds a new `createPoller()` method to the `MessagingProvider` interface with support for additional filter properties.

The filter properties are created as a `Map<String, String>` and can be used to filter the data that is polled from the journal topics.

### CODE REVIEW

**Review by tmaret — COMMENTED** [review]
I suggest to better document the filtering expectations wrt `filterProperties` parameter. What is the semantic of the keys and values in the map ? What logical operation will apply to the filtering (or, and) ?

**Comment by cschneider** [comment]
How about making FilterProperty a simple class instead of creating it via MessagingProvider?
Also I wonder about the need for toMessagingProvider.

**Review by cschneider — APPROVED** [review]
LGTM

File: src/main/java/org/apache/sling/distribution/journal/package-info.java [hunk 1]
```diff
@@ -16,6 +16,6 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-@org.osgi.annotation.versioning.Version("3.1.0")
+@org.osgi.annotation.versioning.Version("1.6.0")
```
Comment (joerghoh) [inline src/main/java/org/apache/sling/distribution/journal/package-info.java]: Why do you downgrade the version of this package?

**Review by joerghoh — CHANGES_REQUESTED** [review]

**Comment by tmaret** [comment]
Could you document the properties semantic in the JavaDoc ?

**Comment by jose-correia** [comment]
@tmaret The API does not demand any specific semantics for these properties. 
They are implemented as a very abstracted String hash table and their semantics specificities depend solely on the API client's implementation of `MessagingProvider`.

Basically the client can use any semantics (depending on his messaging provider design details), and this is completely abstract to this API

**Comment by cschneider** [comment]
I think what we could describe is that the poller will only listen to messages that have properties that match key and name of all the map entries.

**Comment by cschneider** [comment]
Actually it might also make sense to name the properties "filterProperties" to make it more obvious that they are about filtering.

**Comment by tmaret** [comment]
Agreed, initial version was good. IMO we just need to define the parameter better. If the semantic of the keys and values depend on the implementation, let's document that.

**Comment by tmaret** [comment]
Could you also share an example of implementation semantic ?

**Comment by jose-correia** [comment]
@tmaret for example in our specific use-case we run `createPoller()` with the following `filterProperties`:

```json
{
  "publishRegions": "aus"
}
```

Then our `MessagingProvider` implementation knows the exact way for filtering the topics in our specific Journal, and uses this map to create the actual filtering in the journal message:

```json
{
  "properties": [
    {
      "name": "publishRegions",
      "values": ["aus"]
    }
  ],
  "values": {
    ...
  }
}
```

Since the public API is abstracted enough we can support other filtering implementations, it's just a matter of the MessagingProvider creating the final format, since only he knows the exact semantics of the Journal integration it is supporting.

File: src/main/java/org/apache/sling/distribution/journal/MessagingProvider.java [hunk 2]
```diff
@@ -62,11 +62,14 @@ default Closeable createPoller(
      * @param topicName name of the topic
      * @param reset fallback in case no assign is given or the assigned offset not valid
      * @param assign opaque implementation dependent assign string (can be null)
-     * @param properties list of properties to filter the topic
+     * @param filterProperties list of properties used to filter the topic messages.
```
Comment (cschneider) [inline src/main/java/org/apache/sling/distribution/journal/MessagingProvider.java]: I would describe this only in an abstract way like:
List of properties used to filter the topic messages. Only messages that match all filterProperties keys and values will be received.

**Review by cschneider — APPROVED** [review]

**Comment by tmaret** [comment]
Thank you @jose-correia for the example

> Only messages that match all filterProperties keys and values will be received.

+1

**Pull request merged**